core/client/src/main/java/alluxio/client/block/stream/PacketOutStream.java

```diff
@@ -253,24 +253,26 @@ private void updateCurrentPacket(boolean lastPacket) throws IOException {
       return;
     }
 
-    if (mCurrentPacket.readableBytes() > 0) {
-      if (mCurrentPacket.writableBytes() == 0 || lastPacket) {
-        try {
+    if (mCurrentPacket.writableBytes() == 0 || lastPacket) {
```

Comment:
Okay, that makes sense. I thought the last packet would always contain 0 readable bytes.